client/rest-high-level/src/test/java/org/elasticsearch/client/ClusterClientIT.java

```diff
@@ -136,4 +139,29 @@ public void testListTasks() throws IOException {
         }
         assertTrue("List tasks were not found", listTasksFound);
     }
+
+    public void testCancelTasks() throws IOException {
+        ListTasksRequest listRequest = new ListTasksRequest();
+        ListTasksResponse listResponse = execute(
+            listRequest,
+            highLevelClient().cluster()::listTasks,
+            highLevelClient().cluster()::listTasksAsync
+        );
+
+        // TODO[PCS] submit a task that is cancellable and assert it's cancelled
+        // this case is covered in TasksIT.testTasksCancellation
+        TaskInfo firstTask = listResponse.getTasks().get(0);
+        String node = listResponse.getPerNodeTasks().keySet().iterator().next();
+
+        CancelTasksRequest request = new CancelTasksRequest();
+        request.setTaskId(new TaskId(node, firstTask.getId()));
+        request.setReason("persistent action was removed");
+        CancelTasksResponse response = execute(
+            request,
+            highLevelClient().cluster()::cancelTasks,
+            highLevelClient().cluster()::cancelTasksAsync
+        );
+
+        assertThat(response, notNullValue());
```

Comment:
It'd certainly be nice to have more here but you can't really be sure that you are going to actually cancel the task or it is going to finish. Since we already have an integration test for the act of cancelling such a task I'd probably be satisfied with something fairly basic here and good unit testing on the parsing side of the responses. We *know* the cancel works because we have `TaskIT`. Reindex even has some tests to make sure that it cancels itself in a sane way. So I think we're ok if we make sure we can parse here.

I do think it is worth leaving a comment about how this *might* cancel a task and it might not, depending on how quickly the task executes.